### PR TITLE
Module 11 Homework task

### PR DESCRIPTION
**Pull request by dmnikitin**

### CODE REVIEW

File: src/app/courses/components/input-authors/input-authors.component.ts [hunk 1]
```diff
@@ -1,23 +1,97 @@
...
+        }
+      });
+    });
+  }
 
   public ngOnInit(): void {
+    const value: IAuthors[] = this.parent.get('authors').value as IAuthors[];
+    if (value.length) {
+      value.forEach(author => {
+        this.checkedAuthors.push(this.convertFetchedAuthors(author));
```
Comment (KonanMentor) [inline src/app/courses/components/input-authors/input-authors.component.ts:79]: This should be implemented in `writeValue`

File: src/app/courses/pages/add-course-page/add-course-page.component.ts [hunk 2]
```diff
@@ -36,38 +38,64 @@ export class AddCoursePageComponent implements OnInit {
...
         this.router.navigate(['404']);
       }
     });
-    this.form = this.fb.group({
+  }
 
+  public initializeForm(): void {
+    const { name, description, length, date, authors } = this.course;
+    this.form = this.fb.group({
+      name: new FormControl(name, [Validators.required, Validators.maxLength(50)]),
```
Comment (KonanMentor) [inline src/app/courses/pages/add-course-page/add-course-page.component.ts:65]: We may create form group & controls once and update values via `this.form.setValue(this.course)`

File: src/app/courses/components/input-authors/input-authors.component.ts (hunk 1 above)
Comment (dmnikitin) [reply src/app/courses/components/input-authors/input-authors.component.ts]: fixed

File: src/app/courses/pages/add-course-page/add-course-page.component.ts (hunk 2 above)
Comment (dmnikitin) [reply src/app/courses/pages/add-course-page/add-course-page.component.ts]: done

**Pull request merged**